zap/src/test/java/org/zaproxy/zap/utils/ByteBuilderUnitTest.java

```diff
@@ -65,4 +69,314 @@ public void shouldBeInitializedWithDoubleCapacityOfGivenArray() {
         // then
         assertThat(capacity, is(2 * TEST_ARRAY.length));
     }
+
+    @Test
+    public void shouldIncreaseCapacityFrom5To20() {
+        // given
+        byteBuilder = new ByteBuilder(5);
+        // when
+        byteBuilder.ensureCapacity(20);
+        int capacity = byteBuilder.capacity();
+        // then
+        assertThat(capacity, is(20));
+    }
+
+    @Test
+    public void shouldHaveASizeOf0() {
+        // given
+        byteBuilder = new ByteBuilder();
+        // when
+        int size = byteBuilder.size();
+        // then
+        assertThat(size, is(0));
+    }
+
+    @Test
+    public void shouldHaveASizeOf3() {
+        // given
+        byteBuilder = new ByteBuilder(TEST_ARRAY);
+        // when
+        int size = byteBuilder.size();
+        // then
+        assertThat(size, is(3));
+    }
+
+    @Test
+    public void shouldHaveALengthOf3() {
```

Comment:
Same here (and following), it's not clear why it should have a length of 3 (nor the what, it's asserting the length of the string converted to bytes).

Reply:
Test name changed